src/contracts/core/RewardsCoordinator.sol

```diff
@@ -187,25 +196,26 @@ contract RewardsCoordinator is
         PerformanceRewardsSubmission[] calldata performanceRewardsSubmissions
     ) external onlyWhenNotPaused(PAUSED_AVS_PERFORMANCE_REWARDS_SUBMISSION) nonReentrant {
         require(msg.sender == avs, "RewardsCoordinator.createAVSPerformanceRewardsSubmission: caller is not the AVS");
+        // TODO: Should we check if this is a registered AVS?
 
         for (uint256 i = 0; i < performanceRewardsSubmissions.length; i++) {
             PerformanceRewardsSubmission calldata performanceRewardsSubmission = performanceRewardsSubmissions[i];
-            uint256 nonce = submissionNonce[msg.sender];
-            bytes32 performanceRewardsSubmissionHash = keccak256(
-                abi.encode(msg.sender, nonce, performanceRewardsSubmission)
-            );
+            uint256 nonce = submissionNonce[avs];
+            bytes32 performanceRewardsSubmissionHash = keccak256(abi.encode(avs, nonce, performanceRewardsSubmission));
 
             uint256 totalAmount = _validatePerformanceRewardsSubmission(performanceRewardsSubmission);
 
-            isAVSPerformanceRewardsSubmissionHash[msg.sender][performanceRewardsSubmissionHash] = true;
-            submissionNonce[msg.sender] = nonce + 1;
+            isAVSPerformanceRewardsSubmissionHash[avs][performanceRewardsSubmissionHash] = true;
+            submissionNonce[avs] = nonce + 1;
 
             emit AVSPerformanceRewardsSubmissionCreated(
                 msg.sender,
+                avs,
                 nonce,
                 performanceRewardsSubmissionHash,
                 performanceRewardsSubmission
             );
+            // TODO: Figure out if avs or msg.sender should be sender here in the context of delegated accounts.
```

Comment:
prob msg.sender makes the most sense, that way i can delegate the payer for my rewards

Reply:
Yeah I feel the same. The delegated account can approve and call and submit the rewards.

Reply:
Done in [92201fb](https://github.com/Layr-Labs/eigenlayer-contracts/pull/837/commits/92201fb1751d961015aee116b415bef3921d9f39)